lib/flatpak-installation.c

```diff
@@ -1005,20 +1025,51 @@ flatpak_installation_list_remotes (FlatpakInstallation *self,
   if (!flatpak_dir_ensure_repo (dir_clone, cancellable, error))
     return NULL;
 
-  for (i = 0; remote_names[i] != NULL; i++)
+  for (i = 0; i < NUM_FLATPAK_REMOTE_TYPES; ++i)
     {
-      g_ptr_array_add (remotes,
-                       flatpak_remote_new_with_dir (remote_names[i], dir_clone));
+      /* If NULL or an empty array of types is passed then we include all types */
+      types_filter[i] = (num_types == 0) ? TRUE : FALSE;
+    }
+
+  for (i = 0; i < num_types; ++i)
+    types_filter[types[i]] = TRUE;
```

Comment:
Add `g_return_val_if_fail (types[i] < NUM_FLATPAK_REMOTE_TYPES, NULL)` in the loop.